src/isar/state_machine/state_machine.py

```diff
@@ -321,8 +286,39 @@ def _check_dependencies(self):
                 return False
         return True
 
+    def publish_telemetry_thread(self) -> None:
+        robot_id = settings.ROBOT_ID
+        PUBLISH_FREQUENCY = 1
+
+        messenger = Telemetry()
+
+        while True:
+            timestamp = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+            state: IsarStatePayload.State = IsarStatePayload.State(self.current_state)
+            payload_state = IsarStatePayload(state=state, timestamp=timestamp).to_json()
```

Comment:
Could the conversion from `self.current_state` to `state` be done in the constructor of `IsarStatePayload` instead?

Reply:
Yes, it could be done, and agree that would look much better. The reason I don't want to change `IsarStatePayload` for now is that we probably will fetch these models directly from `flotilla-openapi` where they might be auto-generated. I will have this in mind though.